mlflow/spark.py

```diff
@@ -501,7 +508,14 @@ def save_model(
     tmp_path = _tmp_path(dfs_tmpdir)
     spark_model.save(tmp_path)
     sparkml_data_path = os.path.abspath(os.path.join(path, _SPARK_MODEL_PATH_SUB))
-    _HadoopFileSystem.copy_to_local_file(tmp_path, sparkml_data_path, remove_src=True)
+    # If spark DFS is DBFS and we're running on a Databricks cluster, copy to local FS
+    # via the FUSE mount
+    is_saving_to_dbfs = is_valid_dbfs_uri(tmp_path) or posixpath.abspath(tmp_path) == tmp_path
```

Comment:
Good Q, it's because persisting a spark model to a path like "/my-directory" actually saves to "dbfs:/my-directory".

I'll add this explanation/clarifying comment to the code, lmk if it makes sense:

```
    # If we're running on a Databricks cluster, we can detect whether the tempdir we're saving our
    # Spark model to is on DBFS, by checking if it either starts with dbfs:/
    # e.g. "dbfs:/my-directory" or is an absolute path like "/my-directory" (in which case Spark
    # saves to the default distributed FS => DBFS). If we're using a DBFS
    # tempdir, copy to local FS via the FUSE mount. Otherwise, use HDFS APIs to copy the model to
    # the local FS.
```